veps/sig-compute/image-volume-proposal.md

```diff
@@ -0,0 +1,171 @@
+It suggested using shared namespaces between the launcher containers to avoid bind mounts. However, 
+this approach may compromise the isolation of the compute container where the domain runs, as discussed 
+in [here](https://github.com/kubevirt/kubevirt/pull/11845#issuecomment-2413506306).
+
+Another option would be to introduce a new API field to expose the new implementation. However, this would require adding 
+new fields to both the VM and VMI resources and would mean maintaining support for both the old and new implementations 
+for a long time. This would involve updating the CRDs versions, making the upgrade path more complex to support old VMs\VMIs, 
+and removing the old fields. Manual intervention would be required to migrate VMIs to the new approach, and developers 
+would need to adjust many functional tests to the new API. Additionally, this approach would make users aware of implementation 
+details, complicating the API.
+
+The issue with this approach is similar to the problem with the vm.spec.running field, which we’ve been unable to remove
+due to the breaking change it would cause. This demonstrates why adding new API fields to manage the same behavior is not 
+ideal. 
+
+
+## Scalability
+
+This solution can improve KubeVirt VM scalability by reducing the need for extra containers and privileged operations per VM. 
+It will also lower the resources requests and limits by removing the additional containers and their requirements\limits.
+
+## Update/Rollback Compatibility
+
+To make the upgrade path seamless for users, when the feature is graduated, we will temporary continue to support the old 
+method in virt-handler for older launcher pods. We will then migrate the old launcher pods to the new imageVolume method 
+using live migration. This will update the launcher pod spec to use imageVolumes instead of the old sidecar and init 
+containers, and will also update the image to support the new spec.
+
+## Functional Testing Approach
+
+To ensure the new logic works correctly, we will set up a periodic test lane that runs relevant tests with the feature gate 
+enabled. This will allow us to investigate, debug, and confirm that the existing and expected behaviors are functioning 
+as expected until the feature graduates. This will help us build more confidence in the new implementation.
+
+## Implementation Phases
+
+- Add the new implementation, protected by a feature gate that is disabled by default.
+- Once the feature is fully graduated, remove all code related to the old implementation in virt-handler, virt-controller, and virt-launcher, 
+while still supporting the old virt-launcher in virt-handler.
+- During the upgrade to a version where the feature gate is graduated, ensure all launcher pods are updated through the live migration process.
+- After three releases, when we are confident that no launcher pods are using the old logic, remove the remaining code in virt-handler.
+
+## Feature lifecycle Phases
+
+### Alpha
```

Comment:
We need to ensure that Alpha starts only when ImageVolumes are Beta in Kubernetes.

Reply:
Done. I’ve also added this to the proposal: ensure it reaches GA in Kubernetes before moving it to Beta, and enable it by default once it’s in Beta.